sips/sip-x/sip-x-improved-stacking.md

```diff
@@ -0,0 +1,206 @@
+
+- bitcoin block 1-1900: stacking as usual, user can signal change of stacking settings.
+- bitcoin block 1901-2000: stacking as usual, signers can still aggregate stacking changes.
+- bitcoin block 2001-2100 (prepare phase): no rewards for Stackers, changes to stacking are applied to the cycle after next.
+
+```mermaid
+timeline
+    title Stacking Cycle Structure
+    section  Earning
+    1 : Stacking as usual : User can signal change of stacking settings
+    ... : Usual reward distribution
+    1901 : Stacking as usual : Auto extension :  Signers can still aggregate stacking changes
+    section No earning
+    2001 : Prepare phase, no rewards: delegate txs applied only to cycle after next
+    2100 : End of cycle
+```
+
+That means the locking period is 1 cycle, with automatic extension for another cycle until the user decides to end stacking.
+
+## Semantic Change of Delegation
+
+When signers verify and accept proposed blocks by miners, their voting power corresponds to the amount of stacked Stacks tokens (see SIP-021). The new stacking process changes the delegation flow as follows:
+
+- `delegate-stx` shall be renamed to `delegate`/`designate`. The function takes the arguments: `amount`, `signature`, optional bitcoin `block-height` defining the end of stacking and auto extending, optional `pox-addr` defining that pox-address that the signer must use, optional `max-amount`. The `amount` defines how many Stacks tokens are locked immediately. The `max-amount` defines the maximum of Stacks tokens that can be locked in the future through auto extending. If provided the minimum of the user's stx balance and `max-amount` is locked. If omitted, the whole user's balance is locked. Users can use sponsored transactions to call revoke-delegate-stx in case there is no unlocked stx in the account. The argument `signature` is a signature of the signer indicating that the signer accepted the delegation of voting power. The signing follows the structured message signing with the parameters and topic `designate` as message. The public key of the signature can be used to identify the signer similar to the Stacks address of the pool operator in the previous PoX design.
+
+- `revoke-delegate-stx` shall be renamed to `revoke-delegate`/`revoke-designate`. After calling this function, auto extension is stopped and Stacks tokens are unlocked for the user at the end of the current cycle. If a signer does not aggregate enough Stacks tokens to receive at least one reward slot (or does not commit at all) then the user's Stacks token are unlocked immediately through this call.
+
+- `delegate-increase` is replaced by `delegate-update-amount`/`designate-update-amount`. It sets max-amount to the new value. The amount can be smaller than the currently locked amount. As tokens are only locked for 1 cycle at a time, handling decreasing is now easy enough in comparison to the previous system with locking periods of up to 12 cycles.
```

Comment:
Can we change the name here to `update-delegated-amount`?

Reply:
Also, one thing that isn't specified here is whether, on an increase of the `max-amount` from the previous value, whether the STX gets locked immediately.